functions/Test-DbaDbCompression.ps1

```diff
@@ -82,23 +152,105 @@ function Test-DbaDbCompression {
+        [string]$FilterBy = 'Partition',
         [Alias('Silent')]
         [switch]$EnableException
     )
 
     begin {
         Write-Message -Level System -Message "Bound parameters: $($PSBoundParameters.Keys -join ", ")"
+
+        if ($Schema) {
+            $sqlSchemaWhere = "AND s.name IN ('$($Schema -join "','")')"
+        }
+
+        if ($Table) {
+            $sqlTableWhere = "AND t.name IN ('$($Table -join "','")')"
+        }
+
+        if ($MaxResultsPerDb) {
+            $sqlOrderBy = switch ( $RankBy )
+            {
+                UsedPages { 'UsedSpaceKB' }
+                TotalRows { 'RowCounts' }
+                default { 'TotalSpaceKB' }
+            }
+
+            if ($FilterBy -eq 'Table'){
+                $sqlJoinFiltered = 'AND t.TableName = tdc.TableName COLLATE DATABASE_DEFAULT'
+                $indexSQL = '0 as [IndexID]'
+                $partitionSQL = '0 AS [Partition]'
+                $groupBySQL = 's.Name, t.Name'
+            }
+            elseif ($FilterBy -eq 'Index'){
+                $sqlJoinFiltered = 'AND t.TableName = tdc.TableName COLLATE DATABASE_DEFAULT AND t.IndexID = tdc.IndexID'
+                $indexSQL = 'i.index_id as [IndexID]'
+                $partitionSQL = '0 AS [Partition]'
+                $groupBySQL = 's.Name, t.Name, i.index_id'
+            }
+            else {
+                $sqlJoinFiltered = 'AND t.TableName = tdc.TableName COLLATE DATABASE_DEFAULT AND t.IndexID = tdc.IndexID AND t.[Partition] = tdc.[Partition]'
+                $indexSQL = 'i.index_id as [IndexID]'
+                $partitionSQL = 'p.partition_number AS [Partition]'
+                $groupBySQL = 's.Name, t.Name, i.index_id, p.partition_number'
+            }
+
+            $sqlRestrict = "-- remove tables not in Top N
+With TopN(SchemaName, TableName, IndexID, [Partition], RowCounts, TotalSpaceKB, UsedSpaceKB) as
+(
+    SELECT TOP $MaxResultsPerDb
+        s.Name AS SchemaName,
+        t.NAME as TableName,
+        $indexSQL,
+        $partitionSQL,
+        SUM(p.rows) AS RowCounts,
+        SUM(a.total_pages) * 8 AS TotalSpaceKB,
+        SUM(a.used_pages) * 8 AS UsedSpaceKB
+    FROM
+        sys.tables t
+    INNER JOIN
+        sys.indexes i ON t.OBJECT_ID = i.object_id
+    INNER JOIN
+        sys.partitions p ON i.object_id = p.OBJECT_ID AND i.index_id = p.index_id
+    INNER JOIN
+        sys.allocation_units a ON p.partition_id = a.container_id
+    LEFT OUTER JOIN
+        sys.schemas s ON t.schema_id = s.schema_id
+    WHERE objectproperty(t.object_id, 'IsUserTable') = 1
+        AND p.data_compression_desc = 'NONE'
+        AND p.rows > 0
+        $sqlSchemaWhere
+        $sqlTableWhere
+    GROUP BY
+        $groupBySQL
+    ORDER BY
+        $sqlOrderBy Desc
+)
+DELETE tdc
+FROM ##testdbacompression tdc
+LEFT JOIN TopN t
+    ON t.SchemaName = tdc.[Schema] COLLATE DATABASE_DEFAULT
+    $sqlJoinFiltered
+WHERE t.IndexID IS NULL;"
+
```

Comment:
If you don't mind tabbing this text over so it formatts properly, and the code wrapping works correctly in editors.

Reply:
Do you also want the main sql statement adjusted in same manner?